relay-server/src/utils/dynamic_sampling.rs

```diff
@@ -0,0 +1,766 @@
+//! Functionality for calculating if a trace should be processed or dropped.
+//!
+use std::convert::TryInto;
+use std::fmt;
+
+use actix::prelude::*;
+use futures::{future, prelude::*};
+use rand::{distributions::Uniform, Rng};
+use rand_pcg::Pcg32;
+//use serde::{Deserialize, Deserializer, Serialize};
+use serde::de::Visitor;
+use serde::{self, Deserialize, Deserializer, Serialize};
```

Comment:
Since `serde` is a crate, its name is auto-resolved in Edition 2018.

```suggestion
use serde::{Deserialize, Deserializer, Serialize};
```